arangod/GeneralServer/H2CommTask.h

```diff
@@ -0,0 +1,133 @@
+////////////////////////////////////////////////////////////////////////////////
+/// DISCLAIMER
+///
+/// Copyright 2020 ArangoDB GmbH, Cologne, Germany
+///
+/// Licensed under the Apache License, Version 2.0 (the "License");
+/// you may not use this file except in compliance with the License.
+/// You may obtain a copy of the License at
+///
+///     http://www.apache.org/licenses/LICENSE-2.0
+///
+/// Unless required by applicable law or agreed to in writing, software
+/// distributed under the License is distributed on an "AS IS" BASIS,
+/// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+/// See the License for the specific language governing permissions and
+/// limitations under the License.
+///
+/// Copyright holder is ArangoDB GmbH, Cologne, Germany
+///
+/// @author Simon Grätzer
+////////////////////////////////////////////////////////////////////////////////
+
+#ifndef ARANGOD_GENERAL_SERVER_H2_COMM_TASK_H
+#define ARANGOD_GENERAL_SERVER_H2_COMM_TASK_H 1
+
+#include "GeneralServer/AsioSocket.h"
+#include "GeneralServer/GeneralCommTask.h"
+
+#include <nghttp2/nghttp2.h>
+#include <velocypack/StringRef.h>
+#include <boost/lockfree/queue.hpp>
+#include <memory>
```

Comment:
This header file should also include `<map>, `<atomic>` and `<array>`, as it makes use of all of them.